pkg/archive/pom.go

```diff
@@ -0,0 +1,78 @@
+package archive
+
+import (
+	"encoding/xml"
+	"regexp"
+)
+
+type Build struct {
+	Plugins struct {
+		Plugin []struct {
+			ArtifactID    string `xml:"artifactId"`
+			GroupID       string `xml:"groupId"`
+			Configuration struct {
+				// This is the property that contains the name of the git properties file.
+				GenerateGitPropertiesFilename string `xml:"generateGitPropertiesFilename"`
+			} `xml:"configuration"`
+		} `xml:"plugin"`
+	} `xml:"plugins"`
+}
+
+type Project struct {
+	Build Build `xml:"build"`
+
+	Profiles struct {
+		Profile []struct {
+			Build Build `xml:"build"`
+		} `xml:"profile"`
+	} `xml:"profiles"`
+}
+
+func getAllGitPropertiesFileInPom(pom []byte) []string {
+	// This function returns all the git properties files that are generated by git-commit-id-maven-plugin
+	// in the given pom file.
+	// In theory, there can be multiple git properties files in a single pom file so we return a list of them.
+	mavenProject := Project{}
+	err := xml.Unmarshal(pom, &mavenProject)
+	if err != nil {
+		return []string{}
+	}
+	gitPropertiesFiles := getGitPropertiesFilename(mavenProject.Build)
+	for _, profile := range mavenProject.Profiles.Profile {
+		gitPropertiesFiles = append(gitPropertiesFiles, getGitPropertiesFilename(profile.Build)...)
+	}
+	if len(gitPropertiesFiles) == 0 {
+		addDefaultPaths(&gitPropertiesFiles)
+	}
+	return gitPropertiesFiles
+}
+
+func getGitPropertiesFilename(build Build) []string {
+	var gitPropertiesFiles []string
+	for _, plugin := range build.Plugins.Plugin {
+		if plugin.ArtifactID == "git-commit-id-plugin" && plugin.GroupID == "pl.project13.maven" {
+			gitPropertiesFiles = append(gitPropertiesFiles, parsePomProperties(plugin.Configuration.GenerateGitPropertiesFilename))
+		}
+	}
+	return gitPropertiesFiles
+}
+
+func parsePomProperties(property string) string {
```

Comment:
I think this is the only function defined in this file which should remain here given its general name (pom.go). for now, I think its best to move the other functions to the file that defines the git-commit-id stabilizer.